### PR TITLE
4165: support streamfields in service_page imports

### PR DESCRIPTION
**Pull request by Niicck**
# Description
https://github.com/cityofaustin/techstack/issues/4165

- [ ] removes some hardcoded stuff in the importer tests
- [ ] connect tests to a permanent Joplin app for pytest 
- [ ] adds an importer for service pages
- [ ] Makes it optional to create dummy topics and departments for BasePage factories

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 

# Checklist:
- [ ] Request reviewers
- [ ] Slack-out message for review
- [ ] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: joplin/importer/page_importer.py [hunk 1]
```diff
@@ -33,7 +38,7 @@ def fetch_page_data(self):
             # headers={
             #     "Content-type": "application/json",
             # },
-            verify=False
+            verify=True
```
Comment (briaguya) [inline joplin/importer/page_importer.py]: just curious as to what prompted setting verify here, pretty sure it being set to false was pulled from some example somewhere

File: joplin/importer/tests.py [hunk 2]
```diff
@@ -1,70 +1,88 @@
 import pytest
 from importer.page_importer import PageImporter
+from django.core.exceptions import ValidationError
+
 # from unittest.mock import patch
 
 
-def test_parse_janis_preview_url():
-    preview_url = 'https://janis.austintexas.io/en/preview/information/UGFnZVJldmlzaW9uTm9kZToyNjI4'
+def test_parse_janis_preview_url(remote_pytest_preview_url, remote_pytest_api):
```
Comment (briaguya) [inline joplin/importer/tests.py]: this is all so pretty now!

File: joplin/pages/base_page/factories.py [hunk 3]
```diff
@@ -12,16 +12,18 @@ class Meta:
...
+        # TODO: add option to pass in already created departments
         if extracted:
             # A list of departments were passed in, use them
             # for related_department in extracted:
                 # todo: add department stuff here
                 # GroupPagePermissionFactory.create(page=self, topic_collection=topic_collection)
             return
 
-        # todo figure out if this is really what we want this factory to do
+        # pass "add_related_departments__dummy"=True into Factory() to make dummy departments
```
Comment (briaguya) [inline joplin/pages/base_page/factories.py]: good tip on how kwargs work for these

File: joplin/pages/service_page/factories.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        title=page_dictionary['title'],
+        slug=page_dictionary['slug'],
+        add_topics={'topics': topic_pages},
+        add_related_departments=related_departments,
+        coa_global=page_dictionary['coaGlobal'],
+        parent=home,
+        steps=page_dictionary['steps'],
+        dynamic_content=page_dictionary['dynamicContent'],
+        additional_content=page_dictionary['additionalContent'],
+        short_description=page_dictionary['shortDescription'],
```
Comment (briaguya) [inline joplin/pages/service_page/factories.py]: these should be getting cleaner with no camel case!

File: joplin/importer/page_importer.py (hunk 1 above)
Comment (Niicck) [reply joplin/importer/page_importer.py]: It was throwing weird messages during pytest without it. And all of our api endpoints are https, so we're good

**Pull request merged**